### ISSUE TITLE
[Help needed] gspread.exceptions.APIError: {'code': 400, 'message': 'Invalid requests[6].addConditionalFormatRule: Invalid ConditionValue.userEnteredValue: PAST_WEEK', 'status': 'INVALID_ARGUMENT'}

### ISSUE THREAD

**Issue by Jensiii666** [robin900/gspread-formatting#35]
Hey!

This is probably a user related problem, I'm not sure what I'm doing wrong, but thought I'll ask it here because the conditional formatting is not documented that widely for gspread-formatting.

I have:
`rule3 = ConditionalFormatRule(
            ranges=[GridRange.from_a1_range('A23:C23', worksheet)],
            booleanRule=BooleanRule(
                condition=BooleanCondition('DATE_AFTER', ['PAST_WEEK']),
                format=CellFormat(textFormat=textFormat(bold=False), backgroundColor=Color(0.85, 0.92, 0.83))
            )`

I get:
`Traceback (most recent call last):
  File "C:/Users/Joni/PycharmProjects/Oppilasseuranta/Oppilaskorttien päivittäminen.py", line 156, in <module>
    rules.save()
  File "C:\Users\Joni\PycharmProjects\Oppilasseuranta\venv\lib\site-packages\gspread_formatting\conditionals.py", line 83, in save
    resp = self.worksheet.spreadsheet.batch_update(body)
  File "C:\Users\Joni\PycharmProjects\Oppilasseuranta\venv\lib\site-packages\gspread\models.py", line 146, in batch_update
    'post', SPREADSHEET_BATCH_UPDATE_URL % self.id, json=body
  File "C:\Users\Joni\PycharmProjects\Oppilasseuranta\venv\lib\site-packages\gspread\client.py", line 73, in request
    raise APIError(response)
gspread.exceptions.APIError: {'code': 400, 'message': 'Invalid requests[6].addConditionalFormatRule: Invalid ConditionValue.userEnteredValue: PAST_WEEK', 'status': 'INVALID_ARGUMENT'}
`

Is there something I'm just doing wrong and not seeing?

**Comment by robin900** [comment]
The value ‘PAST_WEEK’ is not a userEnteredValue, but instead a relativeDate:

https://www.any-api.com/googleapis_com/sheets/docs/Definitions/BooleanCondition

>

**Comment by robin900** [comment]
So try 'DATE_AFTER', relativeDate='PAST_WEEK'

**Comment by robin900** [comment]
@Jensiii666 Did that work for you?

**Comment by Jensiii666** [comment]
Thank you for responding so quickly. 

Tried your suggestion, I got this:
`Traceback (most recent call last):
  File "C:/Users/Joni/PycharmProjects/Oppilasseuranta/Oppilaskorttien päivittäminen.py", line 55, in <module>
    condition=BooleanCondition('DATE_AFTER', relativeDate='PAST_WEEK'),
TypeError: __init__() got an unexpected keyword argument 'relativeDate'`

saw `class RelativeDate(FormattingComponent)` in conditionals.py and also tried with `RelativeDate=,` but got the exact same type error.

So this was my condition now: `condition=BooleanCondition('DATE_AFTER', relativeDate='PAST_WEEK')`

**Comment by robin900** [comment]
@Jensiii666 I'm so sorry not to address your follow-up questions until now. I gave you an incorrect answer the first time. According to the Sheets API, the correct code would be:
```
rule3 = ConditionalFormatRule( ranges=[GridRange.from_a1_range('A23:C23', worksheet)], booleanRule=BooleanRule( condition=BooleanCondition('DATE_AFTER', [ConditionValue(relativeDate=RelativeDate('PAST_WEEK'))]), format=CellFormat(textFormat=textFormat(bold=False), backgroundColor=Color(0.85, 0.92, 0.83)) )
```
You will notice the _very_ verbose `[ConditionValue(relativeDate=RelativeDate('PAST_WEEK'))]` expression. I've just added more graceful support for relative date values in the BooleanCondition constructor, so that you can instead write:
```
rule3 = ConditionalFormatRule( ranges=[GridRange.from_a1_range('A23:C23', worksheet)], booleanRule=BooleanRule( condition=BooleanCondition('DATE_AFTER', [RelativeDate('PAST_WEEK')]), format=CellFormat(textFormat=textFormat(bold=False), backgroundColor=Color(0.85, 0.92, 0.83)) )
```
and it will be interpreted the same as the more verbose `ConditionValue` expression.

I've committed this change (with unit and integration test coverage) and published a new release 1.0.4 to PyPI. Please let me know if it works for you or if you have any further questions.
